gmdb2: new port

evanmiller opened the pull request:
#### Description

[gmdb2](https://github.com/mdbtools/gmdb2) is the official GUI for [mdbtools](https://github.com/mdbtools/mdbtools) for inspecting and exporting Access database files. Prior to version 0.9 it was included in the main mdbtools distribution, but is now a separate project built on GTK+3.

gmdb2 is currently being packaged for [FreeBSD](https://www.freshports.org/databases/gmdb2/) and in former days was packaged for [Debian](https://packages.debian.org/stretch/mdbtools-gmdb).

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'printf "%s\n" "macOS `sw_vers -productVersion` `sw_vers -buildVersion` `uname -m`" "`xcodebuild -version|awk '\''NR==1{x=$0}END{print x" "$NF}'\''`"'|tee /dev/tty|pbcopy
-->
macOS 10.4.11 8S165 Power Macintosh
Component versions: DevToolsCore-798.0; DevToolsSupport-794.0

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

reneeotten commented:
> My Tiger test machine doesn't support trace (IIRC) – my apologies, I was just doing regular installs.
> 
> I know that autoconfing the project requires `yelp-tools` but it sounds like there is a configure-time dependency as well that I wasn't aware of. I will investigate tomorrow.

oh, that's unfortunate - I didn't realize that this doesn't work on all systems. It's a very useful thing to make sure you're not missing any dependencies and/or use ones that might have been installed already.

I looked a bit more and to me it seems that besides the `itstool` you'd need a build dependency on `pkgconfig` as well. The resulting binary links with files from the `gettext` port, but this is only pulled due to the `itstool` build dependency so MacPorts will allow that one to be uninstalled. To avoid issue because of that it would be good to add the `gettext` port explicitly to `depends_lib`.

With these changes the port does install, but once started it throws this error for me:
```
/opt/local/bin/gmdb2
dbus[33539]: Dynamic session lookup supported but failed: launchd did not provide a socket path, verify that org.freedesktop.dbus-session.plist is loaded!

(gmdb2:33539): GLib-GIO-ERROR **: 13:08:59.979: Settings schema 'mdbtools.gmdb2' is not installed
zsh: trace trap  /opt/local/bin/gmdb2
```

not sure what that's about exactly, but perhaps you've seen that already and know how to fix it?

evanmiller commented:
> not sure what that's about exactly, but perhaps you've seen that already and know how to fix it?

It looks like the dbus subsystem needs to be installed and started in order to access application settings. It's provided through the `dbus` port, but IIRC that starting it up and adding it to LaunchAgents requires an extra step. (I now remember that I went through this process several months ago but have forgotten the details.) I'll need to poke around other Gnome ports and see how they handle this.

Will add the extra build dependencies that you mention.

reneeotten commented:
> It looks like the dbus subsystem needs to be installed and started in order to access application settings. It's provided through the `dbus` port, but IIRC that starting it up and adding it to LaunchAgents requires an extra step.

okay, that does make sense. I briefly looked/tried before my previous comment what is stated in the notes of the `dbus` port:
```
% port notes dbus
--->  dbus has the following notes:
  Startup items (named 'dbus-system, dbus-session') have been generated that will aid in starting
  dbus with launchd. They are disabled by default. Execute the following command to start them, and
  to cause them to launch at startup:

      sudo port load dbus
```

but that didn't do it for me. If there is some manual step required by the user to get it to work (which is fine) then adding a description of that in the `notes` field would be good.

evanmiller commented:
I've added the suggested dependencies and added a line (very common in the Gnome ports) to re-index the GLib application schemas after activation. So that should fix the `GLib-GIO-ERROR`, at least.

I'll need to research dbus some more as I believe it's a separate issue.

reneeotten commented on gnome/gmdb2/Portfile:
```diff
@@ -0,0 +1,32 @@
+                with the .mdb and .accdb extensions, letting the user run queries, \
+                export data, and view a list of stored macros.
+
+homepage        https://github.com/mdbtools/gmdb2
```
it looks like the "name" and "homepage" fields made an unintenional comeback after a recent force-push?

evanmiller replied:
Ah yes, the joys of trying to keep things properly squashed. Will refix.

reneeotten commented:
sounds good - I can give it a try tomorrow. If the initial error is resolved by this it might be all good. 

I think I've seen dbus related messages before without an actual obvious detrimental effect on functionality. So perhaps we can just give it a try like this first, unless you think the software will not function properly of course.

evanmiller commented:
I don't see other ports doing anything special with dbus. It's included / required by `GApplication` in GTK+3. The application itself does not intentionally use it, so it might just be a source of annoying warnings.

reneeotten commented:
> I don't see other ports doing anything special with dbus. It's included / required by `GApplication` in GTK+3. The application itself does not intentionally use it, so it might just be a source of annoying warnings.

@evanmiller thanks - with your latest changes it builds fine and the program opens, so that looks all good! I'd be happy to merge this - anythying else you want to look at beforehand or is this good-to-go?!

evanmiller commented:
@reneeotten Just to be certain, would you mind using the "Open" dialog to choose a file? Just want to make sure dbus isn't needed for that – I do remember seeing some "Open" crashes earlier, which I had attributed to an outdated GLib but never properly diagnosed.

A couple of test files can be found here:

https://github.com/mdbtools/mdbtestdata/tree/master/data

If you can see a list of database tables then I say it's good to merge :-)

One last hiccup is that the Help system requires Yelp – I didn't list this as a dependency since Yelp requires webkit2-gtk, a many-headed Hydra that I am currently wrestling. Would Yelp as an "optional dependency" be worth adding a note about?

reneeotten commented:
opening these test files works fine. It just prints on the terminal many variants of message like:
```
(gmdb2:23043): Gtk-WARNING **: 13:12:18.482: Error loading theme icon 'document-properties' for stock: Icon 'document-properties' not present in theme Adwaita
``` 

You could indeed add a note in the Portfile about that dependency for the help system and that it, therefore, currently doesn't work. If you think that's really relevant and you want to tell the end-user about it, you should use the `notes` field instead - as the user will likely never read the Portfile.

evanmiller commented:
I will look into that warning, maybe it's an outdated or absent `adwaita-icon-theme` port? I should add that port to `depends_run` like other Gnome apps.

I see that other Gnome ports are including `yelp` in `depends_run`. The gmdb2 Helpfile is out of date and not all that useful, so I think instead I'll just use the "notes" field as you suggest.

Will post here when I have these last fixes ready.

evanmiller commented:
I went ahead and added `yelp` and `adwaita-icon-theme` as hard run-time requirements. As much as I'd like to believe there are hordes of PowerPC users trying to read Access databases while using flip phones and listening to Outkast, I think it's easier for everyone just to have the same modern run-time requirements as other Gnome ports.

If CI likes it then it's good to go from my end.

reneeotten commented:
@evanmiller oh, I thought I had merged this already... sorry!

evanmiller commented:
@reneeotten Glad to see it!

(I first downloaded MacPorts just to get gmdb2 and glade building on my machine... didn't realize how addicting it would be :-)
